examples/restaurant_visits/run_without_frameworks_utility_analysis.py

```diff
@@ -63,42 +56,91 @@ def main(unused_argv):
         privacy_id_extractor=lambda row: row.user_id,
         value_extractor=lambda row: row.spent_money)
 
+
+def per_partition_utility_analysis():
+    # Here, we use a local backend for computations. This does not depend on
+    # any pipeline framework and it is implemented in pure Python in
+    # PipelineDP. It keeps all data in memory and is not optimized for large data.
+    # For datasets smaller than ~tens of megabytes, local execution without any
+    # framework is faster than local mode with Beam or Spark.
+    backend = pipeline_dp.LocalBackend()
+
+    # Define the privacy budget available for our computation.
+    budget_accountant = pipeline_dp.NaiveBudgetAccountant(total_epsilon=1,
+                                                          total_delta=1e-6)
+
+    restaurant_visits_rows = load_data(FLAGS.input_file)
+
+    # Create a UtilityAnalysisEngine instance.
+    utility_analysis_engine = UtilityAnalysisEngine(budget_accountant, backend)
+
+    # Create aggregate_params, data_extractors and public partitions.
+    aggregate_params = get_aggregate_params()
+    data_extractors = get_data_extractors()
     public_partitions = list(range(1, 8)) if FLAGS.public_partitions else None
 
-    dp_result = utility_analysis_engine.aggregate(restaurant_visits_rows,
-                                                  params, data_extractors,
-                                                  public_partitions)
+    result = utility_analysis_engine.aggregate(restaurant_visits_rows,
+                                               aggregate_params,
+                                               data_extractors,
+                                               public_partitions)
 
     budget_accountant.compute_budgets()
 
     # Here's where the lazy iterator initiates computations and gets transformed
     # into actual results
-    dp_result = list(dp_result)
+    result = list(result)
 
     # Save the results
-    write_to_file(dp_result, FLAGS.output_file)
+    write_to_file(result, FLAGS.output_file)
+
+
+def aggregate_utility_analysis():
+    # Load data
+    restaurant_visits_rows = load_data(FLAGS.input_file)
+    # Create aggregate_params, data_extractors and public partitions.
+    aggregate_params = get_aggregate_params()
+    data_extractors = get_data_extractors()
+    public_partitions = list(range(1, 8)) if FLAGS.public_partitions else None
+    options = utility_analysis_new.utility_analysis.UtilityAnalysisOptions(
+        1, 1e-5, aggregate_params)
+
+    result = utility_analysis_new.utility_analysis.perform_utility_analysis(
+        restaurant_visits_rows, pipeline_dp.LocalBackend(), options,
+        data_extractors, public_partitions)
+
+    # Here's where the lazy iterator initiates computations and gets transformed
+    # into actual results
+    result = list(result)
+
+    # Save the results
+    write_to_file(result, FLAGS.output_file)
+
 
+def main(unused_args):
```

Comment:
IMO it would be a good idea to include the parameter tuning part here as well when it is ready